pkg/dnsman2/dns/provider/handler/cloudflare/access.go

```diff
@@ -0,0 +1,209 @@
+// SPDX-FileCopyrightText: SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package cloudflare
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudflare/cloudflare-go/v6"
+	cloudflaredns "github.com/cloudflare/cloudflare-go/v6/dns"
+	"github.com/cloudflare/cloudflare-go/v6/option"
+	cloudflarezones "github.com/cloudflare/cloudflare-go/v6/zones"
+	"k8s.io/client-go/util/flowcontrol"
+	"k8s.io/utils/ptr"
+
+	"github.com/gardener/external-dns-management/pkg/dnsman2/dns"
+	"github.com/gardener/external-dns-management/pkg/dnsman2/dns/provider"
+	"github.com/gardener/external-dns-management/pkg/dnsman2/dns/provider/raw"
+)
+
+type accessItf interface {
```

Comment:
Even though this interface has a limited scope, I'm not too happy about the name. What do you think about the following instead?
```suggestion
type accessor interface {
```